File: src/main/java/Inputs.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.List;

public class Inputs {
    private static Path dayToPath(int day) {
        return Paths.get("inputs", String.format("%02d.txt", day));
    }

    public static String readInput(int day) {
        final Path path = dayToPath(day);

        try {
            return Files.readString(path);
        } catch (IOException e) {
            System.out.println("Input for day " + day + " could not be read: " + e);
            System.exit(1);
            throw new RuntimeException();
        }
    }

    public static List<String> readInputLines(int day) {
        final Path path = dayToPath(day);

        try {
            return Files.readAllLines(path);
        } catch (IOException e) {
            System.out.println("Input for day " + day + " could not be read: " + e);
            System.exit(1);
            throw new RuntimeException();
        }
    }
}

File: src/main/kotlin/Inputs.kt
import kotlin.io.path.Path
import kotlin.io.path.exists
import kotlin.io.path.readLines
import kotlin.io.path.readText
import kotlin.system.exitProcess

private fun dayToPath(day: Int) = Path("inputs", "${day.toString().padStart(2, '0')}.txt")
    .also { if (!it.exists()) println("Warning: the input for day $day does not exist!") }

private inline fun <T> tryReadInput(block: () -> T) = runCatching(block)
    .onFailure { println("Failed reading your puzzle input: $it") }
    .getOrElse { exitProcess(1) }

// Reads the input as a large string of text
fun readInput(day: Int) = tryReadInput { dayToPath(day).readText() }

// Reads the input as a list of strings, where each element of the list is a single line of the input file
fun readInputLines(day: Int) = tryReadInput { dayToPath(day).readLines() }